src/main/resources/io/viash/runners/nextflow/workflowFactory/workflowFactory.nf

```diff
@@ -185,44 +189,95 @@ def workflowFactory(Map args, Map defaultWfArgs, Map meta) {
         output_ = output_.findAll{k, v -> k != "_meta"}
 
         // check value types
-        output_ = _processOutputValues(output_, meta.config, id_, key_)
-
-        // simplify output if need be
-        if (workflowArgs.auto.simplifyOutput && output_.size() == 1) {
-          output_ = output_.values()[0]
-        }
+        output_ = _checkValidOutputArgument(output_, meta.config, id_, key_)
 
         [join_id, id_, output_]
       }
       // | view{"chInitialOutput: ${it.take(3)}"}
 
     // join the output [prev_id, new_id, output] with the previous state [prev_id, state, ...]
-    def chNewState = safeJoin(chInitialOutput, chRunFiltered, key_)
+    def chPublishWithPreviousState = safeJoin(chInitialOutputProcessed, chRunFiltered, key_)
       // input tuple format: [join_id, id, output, prev_state, ...]
       // output tuple format: [join_id, id, new_state, ...]
       | map{ tup ->
         def new_state = workflowArgs.toState(tup.drop(1).take(3))
         tup.take(2) + [new_state] + tup.drop(4)
       }
-
     if (workflowArgs.auto.publish == "state") {
-      def chPublish = chNewState
+      def chPublishFiles = chPublishWithPreviousState
         // input tuple format: [join_id, id, new_state, ...]
         // output tuple format: [join_id, id, new_state]
         | map{ tup ->
           tup.take(3)
         }
 
-      safeJoin(chPublish, chArgsWithDefaults, key_)
+      safeJoin(chPublishFiles, chArgsWithDefaults, key_)
         // input tuple format: [join_id, id, new_state, orig_state, ...]
         // output tuple format: [id, new_state, orig_state]
         | map { tup ->
           tup.drop(1).take(3)
+        }
+        | publishFilesByConfig(key: key_, config: meta.config)
+    }
+    def chJoined = chInitialOutputProcessed
+      | groupTuple(by: 1, sort: 'hash', size: chInitialOutputList.size(), remainder: true)
```

Comment:
Question that needs answering: could this `sort` be dropped? This `groupTuple` joins the events across multiple channels before they are passed to `toState`. Since these events are emitted by the channels in random order, do we actually want to sort them? It could be argued that unsorted state is expected and if sorted state is required, it should be handled with `toState`?

Reply:
Also, will removing the `sort` break resume functionality?

Reply:
usually we sort when we want to increase the likelihood that caching through `-resume` will work -- does this apply here?